src/checkers/inference/model/ConstraintManager.java

```diff
@@ -129,6 +130,15 @@ public ExistentialConstraint createExistentialConstraint(Slot slot,
                 ifNotExistsConstraints, getCurrentLocation());
     }
 
+    /**
+     * Create an {@link ArithmeticConstraint} for the given operation and slots.
+     */
+    public ArithmeticConstraint createArithmeticConstraint(ArithmeticOperationKind operation,
+            Slot leftOperand, Slot rightOperand, ArithmeticVariableSlot result) {
+        return ArithmeticConstraint.create(operation, leftOperand, rightOperand, result,
+                getCurrentLocation());
```

Comment:
Did you test if `getCurrentLocation()` really works?

Reply:
For your example above, this is the debug printout:

```Created ArithmeticVariableSlot
  14
  :AstPath: AstPathLocation( A.test1()V.null:A:test1()V::Method.body, Block.statement 0, Variable.initializer )

  20
  :AstPath: AstPathLocation( A.test2()V.null:A:test2()V::Method.body, Block.statement 0, Variable.initializer )
```

So it does work, like in all other `createXXXConstraint` methods in ConstraintManager

Reply:
Hmm, we never cared about the location of a constraint because we don't cache or filter based on the location. The slot has a location for sure, but the constraint manager provides a location everywhere for constraints ...

Reply:
I've updated the ToString serializer to print out the AST path stored in the constraint (from `getCurrentLocation`), and it does have the correct location:

```
Created these Constraints:
  14 = ( 8 @m + 8 @m )
  AstPath: AstPathLocation( A.test1()V.null:A:test1()V::Method.body, Block.statement 0, Variable.initializer )
```